pelican/generators.py

```diff
@@ -749,9 +759,12 @@ def __init__(self, *args, **kwargs):
 
     def generate_context(self):
         self.staticfiles = []
-        for f in self.get_files(self.settings['STATIC_PATHS'],
-                                exclude=self.settings['STATIC_EXCLUDES'],
-                                extensions=False):
+        linked_files = [os.path.join(self.path, path)
+                        for path in self.context['static_links']]
+        found_files = self.get_files(self.settings['STATIC_PATHS'],
+                                     exclude=self.settings['STATIC_EXCLUDES'],
+                                     extensions=False)
+        for f in chain(linked_files, found_files):
```

Comment:
There seems to be a possibility of processing same file twice (thus emitting signals twice etc).

Reply:
You're right, let me look at that.